From abd03df9dec9ac94d29b313b370c97a66db9bdfb Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Tue, 18 Feb 2014 10:36:15 +0100
Subject: [PATCH 25/30] BUG/MINOR: config: server on-marked-* statement is
 ignored in default-server

Commits e0d1bfb ("[MINOR] Allow shutdown of sessions when a server
becomes unavailable") and eb2c24a ("MINOR: checks: add on-marked-up
option") mentionned that the directive was supported in default-server
but while it can be stated there, it's ignored because the config value
is not copied from the default server upon creation of a new server.
Moving the statement to the "server" lines works fine though. Thanks
to Baptiste Assmann for reporting and diagnosing this bug.

These features were introduced in 1.5-dev6 and 1.5-dev10 respectively,
so no backport is needed.
---
 src/cfgparse.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/cfgparse.c b/src/cfgparse.c
index a24eda8..88e86d8 100644
--- a/src/cfgparse.c
+++ b/src/cfgparse.c
@@ -4683,6 +4683,8 @@ stats_error_parsing:
 			newsrv->maxconn		= curproxy->defsrv.maxconn;
 			newsrv->slowstart	= curproxy->defsrv.slowstart;
 			newsrv->onerror		= curproxy->defsrv.onerror;
+			newsrv->onmarkeddown    = curproxy->defsrv.onmarkeddown;
+			newsrv->onmarkedup      = curproxy->defsrv.onmarkedup;
 			newsrv->consecutive_errors_limit
 						= curproxy->defsrv.consecutive_errors_limit;
 #ifdef OPENSSL
-- 
1.8.3.2

